lib/codeowners/cli/main.rb

```diff
@@ -77,10 +77,13 @@ def create_checker(repo)
       }.freeze
 
       def report_errors!(checker)
-        checker.fix!.each do |error_type, inconsistencies|
-          next if inconsistencies.empty?
-
-          puts LABELS[error_type], '-' * 30, inconsistencies, '-' * 30
+        checker.fix!.reduce(nil) do |prev_error_type, (error_type, inconsistencies, meta)|
```

Comment:
I like the idea of de-noising the output by grouping entries of the same type.
Did you consider something like `checker.fix!.group_by` to really group them and maybe sort by count or something like that. I think this would avoid repetition and reduct the number of delimiter `puts` statements. 
Just a thought 🤔

Reply:
so this `#reduce` is actually used as a `#group_by`, at the certain step it was implemented with it but as `#group_by`. But I decided to go with `#reduce` because we are now got results in lazy manner so we can print error immediately once it occurs, I'm not sure that it gives a big benefit there but as it wasn't hard to do I picked a more performant solution. As for number of delimiters, it will be always the same as we output the same error types in raw.
```
 # reduce prints delimiter once prev_error_type != error_typ
:invalid_owner, ... # print delimiter here
:invalid_owner, ...
:invalid_owner, ...
:invalid_pattern, ... # print delimiter here
:invalid_pattern, ...
:missing_ref, ... # print delimiter here
```

So generally the question here is which solution we will prefer:

```
# Lazy and outputs result as soon as way occurs
checker.fix!.reduce(nil) do |prev_error_type, (error_type, inconsistencies, meta)|
  print_delimiter_line(error_type) if prev_error_type != error_type
  case error_type
  when :invalid_owner then puts("#{inconsistencies} MISSING: #{meta.join(', ')}")
  else puts(inconsistencies.to_s)
  end
  error_type
end

# Collects all errors, groups them and prints
checker.fix!.group_by {|(error_type)| error_type}.each do |error_type, group|
  print_delimiter_line(error_type)
  group.each do |(_, inconsistencies, meta)|
    case error_type
    when :invalid_owner then puts("#{inconsistencies} MISSING: #{meta.join(', ')}")
    else puts(inconsistencies.to_s)
    end
  end
end
```

Is pretty hard decision(at least for me) to select the one :)